community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/RelationshipRecord.java

```diff
@@ -238,13 +238,15 @@ public boolean equals( Object o )
                secondPrevRel == that.secondPrevRel &&
                secondNextRel == that.secondNextRel &&
                firstInFirstChain == that.firstInFirstChain &&
-               firstInSecondChain == that.firstInSecondChain;
+               firstInSecondChain == that.firstInSecondChain &&
+               this.requiresSecondaryUnit() == that.requiresSecondaryUnit() &&
```

Comment:
oh, btw I wonder why we have this equals/hashCode in RelationshipRecord, the other records don't. I think it's only for tests.